docgenerator/src/main/scala/codacy/codenarc/docs/DocGenerator.scala

```diff
@@ -0,0 +1,317 @@
+  private val subTypes = reflections.getSubTypesOf(classOf[org.codenarc.rule.AbstractAstVisitorRule]).asScala
+
+  /**
+    * Converts from CodeNarc's complexity to Codacy level
+    */
+  def levelFromPriority(priority: Option[String]): Result.Level.Value = priority match {
+    case Some("1") => Result.Level.Err
+    case Some("2") => Result.Level.Warn
+    case Some("3") => Result.Level.Info
+    case _ => Result.Level.Info
+  }
+
+  def categoryTypeFromPriorityType(priorityType: Option[String]): Pattern.Category.Value = priorityType match {
+    case Some("security") => Pattern.Category.Security
+    case Some("unused") => Pattern.Category.UnusedCode
+    case _ => Pattern.Category.CodeStyle
+  }
+
+  /**
+    * Convert from list of parameters to Parameter.Specification list
+    */
+  def parameterSpecificationFromParametersStringList(
+      parameters: Option[Set[String]]
+  ): Option[Set[Parameter.Specification]] =
+    parameters.map(
+      paramList => paramList.map(param => Parameter.Specification(Parameter.Name(param), Parameter.Value("")))
+    )
+
+  def parameterDescriptionFromParametersStringList(
+      parameters: Option[Set[String]]
+  ): Option[Set[Parameter.Description]] =
+    parameters.map(
+      paramList => paramList.map(param => Parameter.Description(Parameter.Name(param), Parameter.DescriptionText("")))
+    )
+
+  /**
+    * returns the regex to get information from markdown file for a rule
+    *
+    * @param ruleName Name of the rule to get information
+    * @return
+    */
+  def ruleInfoRegex(ruleName: String): Regex = s"## $ruleName([\\n\\S\\s\\w\\d][^##])*".r
+
+  /**
+    * Gets the tool version from the pattern.json file
+    *
+    * @param args
+    * @return
+    */
+  def versionFromArgs(args: Array[String]): String =
+    args.headOption
+      .getOrElse({
+        Using.resource(io.Source.fromFile(codeNarcVersionFile)) { sourceFile =>
+          sourceFile.getLines.mkString("")
+        }
+      })
+
+  /**
+    * Gets the documentation markdown name from the href element value
+    *
+    * @param href
+    * @return
+    */
+  def extractMdFilenameFromHref(href: String): String =
+    href.replaceAllLiterally("./", "").replaceAllLiterally(".html", "").split("#")(0) + ".md"
+
+  /**
+    * Get information about a rule from markdown file
+    *
+    * @param docsFolder   Documentation folder
+    * @param ruleName     Name of the rule to get information about
+    * @param ruleFilename The name of the markdown file where rule information is stored at.
+    * @return
+    */
+  def getRuleExtendedInfoFromMarkdown(docsFolder: String, ruleName: String, ruleFilename: String): String = {
+    val ruleInfoMarkdown = (docsFolder / ruleFilename).contentAsString
+
+    val regexPattern = ruleInfoRegex(ruleName)
+    val regexMatch = regexPattern.findFirstIn(ruleInfoMarkdown).getOrElse("")
+    regexMatch.substring(regexMatch.indexOf(System.lineSeparator) + 1) // remove the title
```

Comment:
Isn't a way to remove the title without using indices in the string?

Reply:
Like `stripPrefix`

Reply:
stripPrefix does not work.
Used linesIterator to remove the title instead